### PR TITLE
fix: union block value

### PR DESCRIPTION
**Pull request by ChenRen93**

### CODE REVIEW

File: src/page/MainPage.tsx [hunk 1]
```diff
@@ -1,32 +1,48 @@
...
+  trigger: "revoke";
+  text: "Revoke Certificate";
+}
+
+interface CancelType {
+  trigger: "cancel";
+  text: "Cancel Pending Transaction";
+}
+
+type blocksType = IssueType | RevokeType | CancelType;
```
Comment (Nebulis) [inline src/page/MainPage.tsx:23]: Let's use the union on he trigger only, the text doesn't matter as we will not use it as index

```ts
type BlockType = {
  trigger: "issue" | "revoke" | "cancel"
  text:string
}
```

File: src/page/MainPage.tsx [hunk 2]
```diff
@@ -61,19 +77,18 @@ export const MainPage: FunctionComponent = () => {
         })}
       </div>
 
-      {documentStoreStatus && (
+      {documentStoreStatus && block !== cancel && (
```
Comment (Nebulis) [inline src/page/MainPage.tsx:80]: block.trigger === "cancel"

File: src/page/MainPage.tsx [hunk 3]
```diff
@@ -1,32 +1,48 @@
...
+const blocks: blocksType[] = [issue, revoke, cancel];
 
 export const MainPage: FunctionComponent = () => {
   const [documentStoreAddress, setDocumentStoreAddress] = useState("");
   const [documentStoreStatus, setDocumentStoreStatus] = useState(false);
-  const [block, showBlock] = useState({
-    trigger: "issue",
-    text: "Issue Certificates",
-  });
+  const [block, showBlock] = useState(blocks[0]);
```
Comment (Nebulis) [inline src/page/MainPage.tsx:45]:
```ts
const [activeBlock, setActiveBlock] = useState(blocks[0].trigger);
```

Comment (ChenRen93) [reply src/page/MainPage.tsx]: done.

File: src/page/MainPage.tsx (hunk 2 above)
Comment (ChenRen93) [reply src/page/MainPage.tsx]: done.

File: src/page/MainPage.tsx (hunk 1 above)
Comment (ChenRen93) [reply src/page/MainPage.tsx]: done.

**Review by Nebulis — APPROVED** [review]

**Pull request merged**